addons/nuxeo-segment.io-connector/src/test/java/org/nuxeo/segment/io/listener/TestSegmentIOListener.java

```diff
@@ -52,19 +57,54 @@
     @Inject
     protected CoreSession session;
 
+    private List<Map<String, Object>> testData;
+
+    @Before
+    public void before() {
+        testData = ((SegmentIOComponent) service).getTestData();
+    }
+
+    @After
+    public void after() {
+        testData.clear();
+    }
+
     @Test
     public void ensureToHandleEveryEventsInABundle() {
         EventBundle eventBundle = new EventBundleImpl();
-        // Add event with ignored users
+        // Add events with ignored users
         eventBundle.push(new EventImpl("dummyEvent", buildCtx("MyAdministrator")));
         eventBundle.push(new EventImpl("dummyEvent", buildCtx("mysystem")));
-        // Add event with another user
+        // Add events with other users
         eventBundle.push(new EventImpl("dummyEvent", buildCtx("johndoe")));
+        eventBundle.push(new EventImpl("dummyEvent", buildCtx("janedoe")));
 
         SegmentIOAsyncListener listener = new SegmentIOAsyncListener();
         listener.handleEvent(eventBundle);
 
-        assertEquals(1, ((SegmentIOComponent) service).getTestData().size());
+        assertEquals(2, testData.size());
+    }
+
+    @Test
+    public void ensureServerUrlIsPassed() {
+        EventBundle eventBundle = new EventBundleImpl();
+        // Add an event with nuxeo.url not set
+        eventBundle.push(new EventImpl("dummyEvent", buildCtx("johndoe")));
+
+        SegmentIOAsyncListener listener = new SegmentIOAsyncListener();
+        listener.handleEvent(eventBundle);
+
+        assertEquals(1, testData.size());
+        assertEquals("unknown server url", testData.get(0).get("url"));
+
+        // set nuxeo.url
+        System.setProperty("nuxeo.url", "http://mytestserver.com");
```

Comment:
Better to use `WithFrameworkProperty` instead :)

Sample here: https://github.com/nuxeo/nuxeo/blob/2387c0ba417810c3b68fd8d51795b46cdee6f579/modules/platform/nuxeo-mail/src/test/java/org/nuxeo/mail/TestMailSessionBuilder.java#L80

Reply:
👍